lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,180 @@
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    onChangeProp(dataTransferObject.files);
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj: File) {
+    const closeBtns = Array.from(selectedFilesContainerRef.current?.
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+
+    if (files) {
+      const filesArray = Array.from(files);
+      filesArray.map((f) => {
+        if (f === fileObj) {
+          const idx = filesArray.indexOf(f);
+          if (idx === files.length - 1) {
+            // if the selected file is the last in the container, set the focus to the next last dismiss button
+            // if selected file is only file, set focus on the input
+            closeBtns.length === 1 ? inputRef.current?.focus() : closeBtns[idx - 1].focus();
+          }
+          else {
+            // need to explicity set focus on next closeBtn for Safari
+            closeBtns[idx].focus();
+          }
+        }
+      });
+    }
+
+    const dataTransferObject = new DataTransfer();
+    if (inputRef.current) {
+      if (files) {
+        forEach((f: File) => dataTransferObject.items.add(f), without([fileObj], Array.from(files)));
+
+        const normalizedFiles = !dataTransferObject.files.length ? null : dataTransferObject.files;
+        onChangeProp(normalizedFiles);
+      }
+    }
+  }
+
+  useEffect(function() {
+    if (inputRef.current) {
+      if (isFileSelected) {
+        inputRef.current.files = files;
+      }
+      else {
+        // there's no way to clear the list via the files prop, but this does it
+        inputRef.current.value = '';
+      }
+    }
+  }, [files]);
+
+  return (
+    <div ref={ref} className={className} >
+      <div className="nx-multi-file-upload__container">
+        <input ref={inputRef}
+               { ...attrs }
+               disabled={disabled}
+               onChange={onChange}
+               id={inputId}
+               className="nx-multi-file-upload__input"
+               type="file"
+               aria-label={isFileSelected ? `${files.length} files selected` : 'No File Selected'}
+               aria-required={isRequired ?? undefined}
+               aria-invalid={showError || undefined}
+               aria-errormessage={showError ? validationErrorId : undefined}
+               multiple/>
+        {/* Keynav and screenreaders can ignore the button itself in favor of the <input> */}
+        <NxButton type="button"
+                  aria-hidden={true}
+                  disabled={disabled}
+                  variant="tertiary"
+                  onClick={openPicker}
+                  tabIndex={-1}
+                  className="nx-multi-file-upload__select-btn">
+          Add Files
```

Comment:
What's the intent with this text? In the mockups it says "Add Files/Directories".  I know we haven't added the directories feature yet, but when we do will that be always enabled (and this text always set as the mockups currently show)? Or will there be an option to enable directories.  If it's the latter, then I assume the "/Directories" part of the button label does not always apply, and there should be mockups for what the button label looks like when that option is off.  Which would also be the mockups for what it should look like as of this current PR.

Reply:
> What's the intent with this text? In the mockups it says "Add Files/Directories". I know we haven't added the directories feature yet, but when we do will that be always enabled (and this text always set as the mockups currently show)? Or will there be an option to enable directories. If it's the latter, then I assume the "/Directories" part of the button label does not always apply, and there should be mockups for what the button label looks like when that option is off. Which would also be the mockups for what it should look like as of this current PR.

At the time I was just leaving it as 'Add Files' until Directories was added - however you bring up a good point. I did reach out to Jason, and they say that they would like to leave the ability to add directories as optional. Without directories, the button will say 'Add Files', and with directories, it will say 'Add Files/Directories' as shown in the mockups. I will reach out to them to update Kaleido as well.